core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -360,9 +360,10 @@ protected void doSample() {
                     try {
                         // its a listed node, light connect to it...
                         logger.trace("connecting to listed node (light) [{}]", listedNode);
-                        transportService.connectToNodeLight(listedNode);
+                        transportService.connectToNodeLight(listedNode, pingTimeout, !ignoreClusterName);
```

Comment:
Does that mean that we can't be sure that we have the version set properly in transport client? I'll have to do some more reading.